implementations/rust/ockam/ockam_command/src/lib.rs

```diff
@@ -447,3 +449,78 @@ pub(crate) fn replace_hyphen_with_stdin(s: String) -> String {
         s
     }
 }
+
+const ENV_FORCE_COLOR: &str = "ockam_force_color";
+
+fn show_help(help: clap::Error) {
+    use std::env;
+    let mut try_fallback = false;
+    let preferred_pager = env::var_os("PAGER").unwrap_or_else(|| {
+        try_fallback = true;
+        OsString::from("less")
+    });
+
+    if preferred_pager == "false" {
+        use clap::{ColorChoice::*, CommandFactory};
+        let possibly_forced = if env::var_os(ENV_FORCE_COLOR).is_some() {
+            Always
+        } else {
+            Auto
+        };
+
+        help.with_cmd(&OckamCommand::command().color(possibly_forced))
+            .exit();
+    }
+
+    if let Ok(()) = paginate_with(preferred_pager, &help) {
+        return;
+    }
+    if try_fallback {
+        if let Ok(()) = paginate_with(OsString::from("more"), &help) {
+            return;
+        }
+    }
+    paginate_with(OsString::from("false"), &help)
+        .expect("displaying help without pagination should always work");
+}
+
+fn paginate_with(pager: OsString, help: &clap::Error) -> Result<()> {
+    let mut pager_invocation = process::Command::new(&pager);
+    if Path::new(&pager)
+        .file_name()
+        .map_or("", |s| s.to_str().unwrap_or(""))
+        == "less"
+    {
+        pager_invocation.env("LESS", "FRX");
+        // - F: no pagination if the text fits entirely into the window
+        // - R: allow ANSI escapes output formatting
+        // - X: prevents clearing the screen on exit
+        // - using env var in case a lesser `less` poses as `less`
+    }
+    let mut pager_process = pager_invocation.stdin(Stdio::piped()).spawn()?;
+    let pipe = Stdio::from(pager_process.stdin.take().expect("stdin open?"));
+
+    let exit_status = {
+        let mut my_args = std::env::args_os();
+        let my_exe_name = my_args.next().unwrap_or("ockam".into());
+
+        let mut rerun = process::Command::new(my_exe_name);
+        rerun.args(my_args).env("PAGER", "false");
+        use atty::Stream::*;
+        let output_stream = if help.use_stderr() {
+            rerun.stderr(pipe);
+            Stderr
+        } else {
+            rerun.stdout(pipe);
+            Stdout
+        };
+        if atty::is(output_stream) {
+            rerun.env(ENV_FORCE_COLOR, "_");
+        }
+        rerun.status()?.code().unwrap_or(exitcode::SOFTWARE)
+        // dropping owned pipe hands over pager control to the user
+    };
```

Comment:
I'm sure we can do this without having to rerun the same command. Did you need it to add the `ENV_FORCE_COLOR`?

Reply:
Where's my reply? Did I hit the wrong button?

Reply:
That said, I personally don't really really like the above suggestion, since some settings already look crappy using a pager by default. Git doesn't need it. And there is means to control the automatic by connecting the output channels to (possibly pseudo-) terminals or not.

You mentioned a conventional `NO_COLOR` env var. Supporting that one (and keeping the internal one or piggy-backing on `NO_COLOR` with an internal "impossible"-to-collide value that instead forces colors to ON - just for communicating with the child - both cases being mutually exclusive) would be another low-hanging fruit and better aligned with the rest of the existing interface.

Reply:
We already support the `NO_COLOR` env var and we have a way to determine if the streams support color, although we should introduce the internal "force-color" setting as you mentioned. There's definitely room for improvement on this regard. We'll give it a thought after testing this for a while.